rabbitmq/tests/test_openmetrics.py

```diff
@@ -27,274 +44,118 @@ def test_aggregated_endpoint(aggregated_setting, aggregator, dd_run_check, mock_
-                    'unaggregated_endpoint': 'detailed?family=queue_coarse_metrics',
-                    "include_aggregated_endpoint": False,
-                },
-                "metrics": [".+"],
-            }
-        ],
+    ],
+)
+def test_detailed_endpoint(endpoint, fixture_file, expected_metrics, aggregator, dd_run_check, mock_http_response):
+    mock_http_response(file_path=OM_RESPONSE_FIXTURES / fixture_file)
+    check = _rmq_om_check(
+        {
+            'url': TEST_URL,
+            'unaggregated_endpoint': endpoint,
+            "include_aggregated_endpoint": False,
+        }
     )
     dd_run_check(check)
 
-    expected_metrics = (
-        [
-            dict(
-                name='rabbitmq.build_info',
-                value=1,
-                metric_type=aggregator.GAUGE,
-                tags=[
-                    'erlang_version:25.1.2',
-                    'prometheus_client_version:4.9.1',
-                    'prometheus_plugin_version:3.11.3',
-                    'rabbitmq_version:3.11.3',
-                ],
-            ),
-            dict(
-                name='rabbitmq.identity_info',
-                value=1,
-                metric_type=aggregator.GAUGE,
-                tags=[
-                    'rabbitmq_node:rabbit@54cfac2199f1',
-                    "rabbitmq_cluster:rabbit@54cfac2199f1",
-                    "rabbitmq_cluster_permanent_id:rabbitmq-cluster-id-cyw_z6c4UMIBoK51iVq9rw",
-                ],
-            ),
-        ]
-        + [
-            dict(
-                name=name,
-                value=value,
-                metric_type=aggregator.GAUGE,
-                tags=['vhost:/', f'queue:{qname}'],
-            )
-            for name, (value, qname) in product(
-                ('rabbitmq.queue.messages', 'rabbitmq.queue.messages.ready'),
-                ((0, 'queue1'), (1, 'queue2'), (0, 'queue3')),
-            )
-        ]
-        + [
-            dict(
-                name='rabbitmq.queue.messages.unacked',
-                value=0,
-                metric_type=aggregator.GAUGE,
-                tags=['vhost:/', f'queue:{qname}'],
-            )
-            for qname in ('queue1', 'queue2', 'queue3')
-        ]
-        + [
-            dict(
-                name='rabbitmq.queue.process_reductions.count',
-                value=value,
-                metric_type=aggregator.MONOTONIC_COUNT,
-                tags=['vhost:/', f'queue:{qname}'],
-            )
-            for qname, value in (('queue1', 30773), ('queue2', 30466), ('queue3', 35146))
-        ]
-    )
     for m in expected_metrics:
-        kwargs = {
-            **m,
-            "tags": [f"endpoint:{TEST_URL}/metrics/detailed?family=queue_coarse_metrics"] + m.get('tags', []),
-        }
-        aggregator.assert_metric(**kwargs)
+        aggregator.assert_metric(m)
+
+    for m in set(DEFAULT_OPENMETRICS).difference(expected_metrics):
+        aggregator.assert_metric(m, at_least=0)
```

Comment:
Re: `at_least=0`, are these metrics expected to not be present in the fixtures?

Reply:
Oh I see we're checking to make sure those metrics aren't duplicated?

Reply:
Yup! lmk if there's a more idiomatic way to do so :)

Reply:
We can a comment there, like you did for the below test https://github.com/DataDog/integrations-core/pull/13737/files#diff-88b775cd970d634634d15c14d038fea92ec57624a27a75da8da328aabbf9d6f4R189-R190

Reply:
Ah, duplication strikes again. I'll copy/paste the comment for now!